File: Greedy/잃어버린 괄호 - 1541.py
expr = input()
parts = expr.split('-')
# +부분을 모두 먼저 더해서 나중에 빼줌
for i in range(len(parts)):
    plus_parts = list(map(int, parts[i].split('+')))
    total = plus_parts[0]
    for j in range(1, len(plus_parts)):
        total += plus_parts[j]
    parts[i] = total

answer = parts[0]
for i in range(1, len(parts)):
    answer -= parts[i]

print(answer)

File: Greedy/잃어버린 괄호 - 1541.kt
import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val parts = readLine().split('-').toTypedArray()
    // +부분을 모두 먼저 더함 (괄호를 침) 
    for (i in parts.indices) {
        val plus_parts = parts[i].split('+').map{ it.toInt() }.toIntArray()
        parts[i] = plus_parts.sum().toString()
    }
    
    // +한 부분들을 모두 빼줌
    var answer = parts[0].toInt()
    for (i in 1 until parts.size) answer -= parts[i].toInt()
    
    print(answer)
}
